executor/analyze.go

```diff
@@ -345,31 +350,32 @@ func (e *AnalyzeExec) analyzeWorker(taskCh <-chan *analyzeTask, resultsCh chan<-
 				Job: task.job,
 			}
 		}
-		e.wg.Done()
-		if isCloseChanThread {
-			e.wg.Wait()
-			close(resultsCh)
-		}
 	}()
 	for {
-		var ok bool
-		task, ok = <-taskCh
-		if !ok {
-			break
-		}
-		task.job.Start()
-		switch task.taskType {
-		case colTask:
-			resultsCh <- analyzeColumnsPushdown(task.colExec)
-		case idxTask:
-			resultsCh <- analyzeIndexPushdown(task.idxExec)
-		case fastTask:
-			resultsCh <- analyzeFastExec(task.fastExec)
-		case pkIncrementalTask:
-			resultsCh <- analyzePKIncremental(task.colIncrementalExec)
-		case idxIncrementalTask:
-			resultsCh <- analyzeIndexIncremental(task.idxIncrementalExec)
+		select {
+		case task := <-taskCh:
```

Comment:
Can we make worker exit when it finds `taskCh` is closed?

Reply:
removed